scripts/deploy/09_eth_put_theta_vault.ts

```diff
@@ -0,0 +1,163 @@
+  [CHAINID.AVAX_MAINNET]: 'rUSDC-AVAX-P-THETA',
+  [CHAINID.AVAX_FUJI]: 'rUSDC-AVAX-P-THETA',
+};
+
+const STRIKE_STEP = {
+  [CHAINID.ETH_MAINNET]: ETH_STRIKE_STEP,
+  [CHAINID.ETH_KOVAN]: ETH_STRIKE_STEP,
+  [CHAINID.AVAX_MAINNET]: AVAX_STRIKE_STEP,
+  [CHAINID.AVAX_FUJI]: AVAX_STRIKE_STEP,
+};
+
+const main = async ({
+  network,
+  deployments,
+  ethers,
+  getNamedAccounts,
+}: HardhatRuntimeEnvironment) => {
+  const { BigNumber } = ethers;
+  const { deploy } = deployments;
+  const { deployer, owner, keeper, admin, feeRecipient } =
+    await getNamedAccounts();
+  console.log(`09 - Deploying ETH Put Theta Vault on ${network.name}`);
+
+  const chainId = network.config.chainId;
+
+  const manualVolOracle = await deployments.get("ManualVolOracle");
+  const underlyingOracle = ETH_PRICE_ORACLE[chainId];
+  const stablesOracle = USDC_PRICE_ORACLE[chainId];
+
+  const pricer = await deploy("OptionsPremiumPricerETH", {
+    from: deployer,
+    contract: {
+      abi: OptionsPremiumPricerInStables_ABI,
+      bytecode: OptionsPremiumPricerInStables_BYTECODE,
+    },
+    args: [
+      ETH_USDC_POOL[chainId],
+      manualVolOracle.address,
+      underlyingOracle,
+      stablesOracle,
+    ],
+  });
+
+  console.log(`RibbonThetaVaultETHCall pricer @ ${pricer.address}`);
+
+  // Can't verify pricer because it's compiled with 0.7.3
+
+  const strikeSelection = await deploy("StrikeSelectionETH", {
+    contract: "DeltaStrikeSelection",
+    from: deployer,
+    args: [
+      pricer.address,
+      STRIKE_DELTA,
+      STRIKE_STEP[chainId],
+    ],
+  });
+
+  console.log(`RibbonThetaVaultETHCall strikeSelection @ ${strikeSelection.address}`);
+
+  try {
+    await run('verify:verify', {
+      address: strikeSelection.address,
+      constructorArguments: [
+        pricer.address,
+        STRIKE_DELTA,
+        STRIKE_STEP[chainId],
+      ],
+    });
+  } catch (error) {
+    console.log(error);
+  }
+
+  // Assumes these contracts are already deployed
+  const lifecycle = await deployments.get("VaultLifecycle");
+  const logicDeployment = await deployments.get("RibbonThetaVaultLogic");
+  const RibbonThetaVault = await ethers.getContractFactory("RibbonThetaVault", {
+      libraries: { VaultLifecycle: lifecycle.address }
+  });
+
+  const initArgs = [
+    {
+      _owner: owner,
+      _keeper: keeper,
+      _feeRecipient: feeRecipient,
+      _managementFee: MANAGEMENT_FEE,
+      _performanceFee: PERFORMANCE_FEE,
+      _tokenName: TOKEN_NAME[chainId],
+      _tokenSymbol: TOKEN_SYMBOL[chainId],
+      _optionsPremiumPricer: pricer.address,
+      _strikeSelection: strikeSelection.address,
+      _premiumDiscount: PREMIUM_DISCOUNT,
+      _auctionDuration: AUCTION_DURATION,
+      _isUsdcAuction: false,
+      _swapPath: 0x0,
+    },
+    {
+      isPut: true,
+      decimals: 6,
+      asset: USDC_ADDRESS[chainId],
+      underlying: WETH_ADDRESS[chainId],
+      minimumSupply: BigNumber.from("10").mul(BigNumber.from(10).pow(6)),
```

Comment:
We're using 10**3 for our USDC put vaults so far. Reason is that if the minimumSupply is too high, it will revert on small deposits. See the [yvUSDC vault deployment for reference](https://github.com/ribbon-finance/ribbon-v2/blob/ken/deploy-yvusdc-script/scripts/deploy/10_yvusdc_eth_put_theta_vault.ts#L45-L65)

Reply:
Fixed